### PR TITLE
Add --path argument to container:push

### LINKED ISSUE #74: heroku container:push should allow specifying a build context path
`heroku container:push` doens't build the images correctly if the Dockerfile is not on the project root.

I keep Dockerfiles on a separate folder and currently build the images with `docker build -f docker/Dockerfile.web .`

Adding a way to specify the build context path would allow me to build using `heroku container:push --recursive --path .`

### PR DESCRIPTION
**Pull request by eduardocardoso**
Fixes #74

### CODE REVIEW

File: lib/sanbashi.js [hunk 1]
```diff
@@ -80,8 +80,12 @@ Sanbashi.filterByProcessType = function (jobs, procs) {
   return filteredJobs
 }
 
-Sanbashi.buildImage = function (dockerfile, resource, buildArg) {
+Sanbashi.buildImage = function (dockerfile, resource, buildArg, path) {
   let cwd = Path.dirname(dockerfile)
+  if (path) {
+    cwd = Path.join(process.cwd(), path)
```
Comment (dmathieu) [inline lib/sanbashi.js]: Sanbashi shouldn't use `process.cwd()`. If we need an absolute path here, it should be passed upstream.
Also, what if I do pass an absolute path to the command?

File: commands/push.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ module.exports = function (topic) {
       `${cli.color.cmd('heroku container:push worker')}                       # Pushes Dockerfile to worker process type`,
       `${cli.color.cmd('heroku container:push web worker --recursive')}       # Pushes Dockerfile.web and Dockerfile.worker`,
       `${cli.color.cmd('heroku container:push --recursive')}                  # Pushes Dockerfile.*`,
-      `${cli.color.cmd('heroku container:push web --arg ENV=live,HTTPS=on')}  # Build-time variables`
+      `${cli.color.cmd('heroku container:push web --arg ENV=live,HTTPS=on')}  # Build-time variables`,
+      `${cli.color.cmd('heroku container:push --recursive --path .')}         # Pushes Dockerfile.* using current dir as build context`
```
Comment (dmathieu) [inline commands/push.js]: Not sure we should give an example for every option. @heroku/cli wdyt?

File: commands/push.js [hunk 3]
```diff
@@ -33,6 +34,12 @@ module.exports = function (topic) {
         name: 'arg',
         hasValue: true,
         description: 'set build-time variables'
+      },
+      {
+        name: 'path',
```
Comment (dmathieu) [inline commands/push.js]: `--path` is a bit obscure IMHO. I'd name this `--context`, or maybe even `--context-path`.

**Review by dmathieu — COMMENTED** [review]
This is a very good idea, thank you.

File: lib/sanbashi.js (hunk 1 above)
Comment (eduardocardoso) [reply lib/sanbashi.js]: I don't think an absolute path is needed here. I'll just pass whatever comes on --path as an arg to docker build

File: commands/push.js (hunk 3 above)
Comment (eduardocardoso) [reply commands/push.js]: Agreed, will change to `--context-path`

**Comment by dmathieu** [comment]
Could you add a test for the command as well?
https://github.com/heroku/heroku-container-registry/blob/master/test/commands/push.js#L55

**Review by dmathieu — APPROVED** [review]

**Pull request closed without merging**